support/cas-server-support-consent-webflow/src/main/java/org/apereo/cas/web/consent/CasConsentReviewController.java

```diff
@@ -0,0 +1,100 @@
+import org.springframework.web.bind.annotation.ResponseBody;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * This is {@link CasConsentReviewController}.
+ *
+ * @author Arnold Bergner
+ * @since 5.2.0
+ */
+@Controller("casConsentReviewController")
+@RequestMapping("/consent")
+public class CasConsentReviewController {
+    private static final Logger LOGGER = LoggerFactory.getLogger(CasConsentReviewController.class);
+
+    private final ConsentRepository consentRepository;
+    private final ConsentEngine consentEngine;
+
+    public CasConsentReviewController(final ConsentRepository consentRepository, final ConsentEngine consentEngine) {
+        this.consentRepository = consentRepository;
+        this.consentEngine = consentEngine;
+    }
+
+    /**
+     * Show consent decisions.
+     *
+     * @param request  the request
+     * @param response the response
+     * @return the model and view where json data will be rendered
+     */
+    @GetMapping
+    public String showConsent(final HttpServletRequest request,
+                                        final HttpServletResponse response) {
+        return "casConsentReviewView";
+    }
+
+    /**
+     * Endpoint for getting consent decisions in JSON format.
+     *
+     * @param request  the request
+     * @param response the response
+     * @return the consent decisions
+     */
+    @GetMapping("/getConsentDecisions")
+    @ResponseBody
+    public Collection<Map<String, Object>> getConsentDecisions(final HttpServletRequest request,
+                                                            final HttpServletResponse response) {
+        final WebContext context = new J2EContext(request, response);
+        final ProfileManager manager = new ProfileManager(context);
+        final Optional<UserProfile> profile = manager.get(true);
+        if (profile.isPresent()) {
+            final String principal = profile.get().getId();
+            LOGGER.debug("Fetching consent decisions for principal [{}]", principal);
+            final Collection<ConsentDecision> consentDecisions = this.consentRepository.findConsentDecisions(principal);
+            LOGGER.debug("Resolved consent decisions for principal [{}]: {}", principal, consentDecisions);
+            final Collection<Map<String, Object>> result = new HashSet<>();
+            consentDecisions.stream().map(this::decodeDecision).forEach(result::add);
+            return result;
+        }
+        return null;
+    }
+
+    /**
+     * Returns decoded consent decision.
+     * @param decision the decision
+     * @return the decoded decision
+     */
+    private Map<String, Object> decodeDecision(final ConsentDecision decision) {
+        final Map<String, Object> map = new HashMap<>();
+        map.put("id", decision.getId());
```

Comment:
Why not pass the decision and attributes? Why break down to individual fields? If a new field is added in the future, it needs to also be added here right? Passing the decision object entirely helps to reduce work?